contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -745,6 +753,19 @@ contract SmartAccount is
         UserOperation calldata userOp,
         bytes32 userOpHash
     ) internal virtual override returns (uint256 validationData) {
+        // below changes need formal verification.
+        bytes calldata userOpData = userOp.callData;
+        if (userOpData.length > 0) {
+            bytes4 methodSig = bytes4(userOpData[:4]);
+            // If method to be called is executeCall then only check for module transaction
+            if (methodSig == this.executeCall.selector) {
+                (address _to, uint _amount, bytes memory _data) = abi.decode(
+                    userOpData[4:],
+                    (address, uint, bytes)
+                );
+                if (address(modules[_to]) != address(0)) return 0;
```

Comment:
In this case, shouldn't we pass the signature validation flow to the module?

Reply:
the method being called on module indeed does the validation (within it's written logic already), but it may not relate with userOpHash and original owner signature.